addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -299,4 +295,28 @@ protected long lowerThresholdToUseMultipartCopy() {
         return NON_MULTIPART_COPY_MAX_SIZE;
     }
 
+    /** @since 11.1 */
+    protected Credentials getAwsCredentials(String batchId) {
+        AssumeRoleRequest request = new AssumeRoleRequest().withRoleArn(roleArn)
+                                                           .withPolicy(policy)
+                                                           .withRoleSessionName(batchId);
+        if (expiration > 0) {
+            request.setDurationSeconds(expiration);
+        }
+        return assumeRole(request);
+    }
+
+    /** @since 11.1 */
+    public Map<String, Object> refreshToken(String batchId) {
```

Comment:
Even if it's not _required_ you should also have an `@Override` here. The code should be as similar to master as possible except for required changes.

Reply:
Done :)